### ISSUE TITLE
Trying to follow guide to getting started with Hubot, results differ from what the guide says.

### ISSUE THREAD

**Issue by NanoDami** [github/hubot#979]
So i am trying to follow the guide to simply install Hubot on my Ubuntu (14.04) server, but things do not turn out the way the guide says it will do. I've set up my server according to this guide: https://www.linode.com/docs/getting-started

And it looks like everything is ready get started.

Steps.

1. The guide says to follow another guide which says i need a compiler, i guess i can just use "sudo aptitude install build-essential" and i'm good to go?

2. "which gcc" now reveals that i now do have a compiler! (/usr/bin/gcc)

3. I use the first gist to install NPM and node.js and i then have to wait a really long time for all the text in my terminal to settle.

4. There is a line in the gist that did not execute: "curl https://www.npmjs.org/install.sh | sh", and when i try to execute it, i get this message:  
 % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                                       Dload  Upload   Total   Spent    Left  Speed
100    51  100    51    0     0    291      0 --:--:-- --:--:-- --:--:--   293
sh: 1: Moved: not found

I don't really know what it means, but at least i can try to just run the commands in install.sh manually and see what happens, right?

5.So i tried running it, and i got alot of text appearing. Okay, something happened. Woo! But then i get this message:

npm cannot be installed without node.js.
Install node first, and then try again.

Maybe node is installed, but not in the PATH?
Note that running as sudo can change envs.

PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

So it looks like install.sh does not know where my node-files are. Do you have any tips on what i can do further? The guide you have here does not really help that much for new users with little experience.

Aaaaand it looks like something happened to my ssh, making me unable to ssh into my server.

Am i just being to much of a noob to set up Hubot properly or is something in the guides you provide not functioning as intended?

**Comment by technicalpickles** [comment]
> but things do not turn out the way the guide says it will do.

To make sure we are looking at the same guide, can you link me to it?

**Comment by NanoDami** [comment]
Sorry, i forgot to link to that one. 
The guide i am following is the one on your documentation page:
https://hubot.github.com/docs/

**Comment by technicalpickles** [comment]
So the problem isn't with the hubot guide, but the [node and npm blog post](https://www.joyent.com/blog/installing-node-and-npm) linked to from it. And that links to a [gist](https://gist3.github.com/isaacs/579814). The reason for linking to another resource is to avoid having to document it ourselves, but that is not working :sweat: 


If you check out the gist, it actually has comments about this. Use this instead: `curl -L https://www.npmjs.org/install.sh | sh`.

Basically, there's a redirect happening from www.npmjs.org to www.npmjs.com, and `-L` makes it follow the redirect.

**Comment by NanoDami** [comment]
Ah, i see! I will try it out then and report back my findings.
If wanted, i could probably write a summary of what i am doing, with changes and correct steps, and you could review it. If you deem it good enough, it could probably be a nice addition to your already fine documentation. 
And why would you avoid making your own documentation that would ensure users could get the most of  the software as fast as possible? It will make it easier to answer questions when people can cite your own, updated guides instead of following a 5 year old guide.

**Comment by technicalpickles** [comment]
> And why would you avoid making your own documentation that would ensure users could get the most of the software as fast as possible? It will make it easier to answer questions when people can cite your own, updated guides instead of following a 5 year old guide.

I'm all for making hubot easier for people to use. But the reason to not write it ourself because there are a lot of different ways to install node, even with a single platform, so it's hard to provide quality docs even documenting the best ones. There is a lot of work I could be doing on hubot without even touching on installing node. But given this, I'd consider documentation updates splitting out "install node & npm" document that can cite various package managers and whatnot.

**Comment by technicalpickles** [comment]
@NanoDami we've updated the documentation to link to https://docs.npmjs.com/getting-started/installing-node for installing. Can you take a look at that and see if it gets you further?
